File: service/src/main/kotlin/com/yt/service/annotation/RedisCache.java
package com.yt.service.annotation;


import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Target({ElementType.METHOD,ElementType.TYPE})
@Retention(RetentionPolicy.RUNTIME)
public @interface RedisCache {


    Operate operate() default Operate.QUERY;

    String nameSpace() default "";

    long cacheTimeSecond() default 5 * 60L;
}

File: appcommon/src/main/kotlin/com/yt/appcommon/annotation/RedisCache.kt
package com.yt.appcommon.annotation

import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy

@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER, AnnotationTarget.ANNOTATION_CLASS, AnnotationTarget.CLASS)
@Retention(RetentionPolicy.RUNTIME)
annotation class RedisCache(val operate: Operate = Operate.QUERY, val nameSpace: String = "", val cacheTimeSecond: Long = 5 * 60L)